File: Python/2483. Minimum Penalty for a Shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        customers = "X" + customers
        y, n = customers.count("Y"), 0
        r, m = 0, float("inf")

        for i, x in enumerate(customers):
            if x == "Y": y -= 1
            elif x == "N": n += 1

            if y + n < m: m = y + n; r = i

        return r

File: C++/2483. Minimum Penalty for a Shop.cpp
class Solution {
public:
    int bestClosingTime(string customers) {
        customers = "X" + customers;
        int y = 0, n = 0, r = 0, m = 100001;

        for (char i: customers) {
            if (i == 'Y') y++;
        }

        for (int i = 0; i < customers.length(); i++) {
            if (customers[i] == 'Y') y--;
            else n++;

            if (n + y < m) m = n + y, r = i;
        }

        return r;
    }
};
